next/app/lib/components/Navbar.tsx

```diff
@@ -0,0 +1,80 @@
+
+"use client";
+
+import React from 'react';
+import { Routes } from '../constants';
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import { User } from '@auth/core/types';
+import { FaAngleDown, FaAngleUp } from 'react-icons/fa';
+import { keycloakSignOut } from '../actions/keycloak';
+import { Role } from "@/prisma/generated/client";
+
+
+export interface INavbarOption {
+  label: string;
+  route: string;
+  role: Role;
+}
+
+export interface INavbarProps {
+  user: User;
+}
+
+/** Client Component used for navigation */
+export const Navbar: React.FC<INavbarProps> = ({ user }) => {
+  const pathname = usePathname();
+  const [showUserDropDown, setShowUserDropDown] = React.useState(false);
+
+  const navItems: INavbarOption[] = [
+    { label: 'Home', route: Routes.Home, role: Role.ZEVA_USER },
```

Comment:
I have a simple object here that includes what role is needed to view the navigation item; however, I am wondering if there is somewhere I can see what role is actually needed? These are placeholders, I noticed we used `user.hasPermission()` instead of role on the last one. But I am probably missing something :stuck_out_tongue:

Reply:
Hi Jonny, I forgot to mention that, in order to simplify things, I think we should do only roles in zeva2, and not roles + permissions like we do in zeva1. Also, as far as I can tell, there's no way in zeva1 for users to change which permissions are associated with which roles, so it seemed a bit redundant to me to have permissions. 

So, in the original Navbar, if you have something like `user.hasPermission('EDIT SALES')`, then you can look at the `role_permission` product table in the zeva1 database and see that the roles `Engineer/Analyst`, `Signing Authority`, and `ZEVA User` are associated with that permission; so then you would do something like `user.roles.includes(Role.ENGINEER_ANALYST, Role.SIGNING_AUTHORITY, Role.ZEVA_USER)` in place of that `user.hasPermission`.

Reply:
Got it! Thanks @tim738745

Reply:
Updated!